### ISSUE TITLE
[Org] Semester-Master für die Studis anlegen?

### ISSUE THREAD

**Issue by cagix** [Programmiermethoden/Dungeon#650]
Wir hatten mal darüber diskutiert und ich war ursprünglich gegen Versionen und/oder Semesterbranches etc., um die Sache überschaubar zu halten und weil ich tatsächlich nicht wirklich erwartet habe, dass irgendwer seinen/ihren Fork mal aktualisiert. Das scheint nun aber doch zu passieren, und durch einige unserer Refactorings in der letzten (und kommenden) Zeit werden dabei Probleme verursacht, da die Studis teilweise munter im Vorgabecode arbeiten.

Überlegung: Sollten wir vom letzten Commit, der mit dem Stand von Anfang April kompatibel ist, einen Hilfsbranch "pm_s23_master" abzweigen? Dann können wir im eigentlichen Master weiterarbeiten und müssten ggf. Bugfixes dann auch nochmal extra in den Hilfsbranch einpflegen (bisher hatten wir nur einen Bugfix, das lässt hoffen). Am Semesterende fliegt der Hilfsbranch dann wieder raus. Ich würde das Vorgehen auch an die Studis kommunizieren.

Es könnte sein (müsste nochmal geprüft werden), dass a2e780eb320d28b3f25fc35ae3bf849d00b0791d der letzte non-breaking Commit/Merge war. Der nächste Merge (938c5801973eb6dd11e41954bd2eda6b9b18e101) benennt schon `ECS_System` nach `System` um und könnte all denen Schwierigkeiten beim Update machen, die da drin rumgearbeitet haben.

@AMatutat @AHeinisch @fwatermann @JudiTeller Gedanken?

**Comment by AMatutat** [comment]
Zwei Versionen haben immer den faden Geschmack von zwei verschiedenen Bugfixes.
Wir hatten das im alten Dungeon schonmal durchgemacht, war nicht schön.
Aus Studi Sicht verstehe ich natürlich, dass das aktuelle Vorgehen murks ist. 

Kompromiss: Wir erstellen einen Branch wie von dir beschrieben, führen bugfixes aber nur auf Zuruf der Studis aus. Heißt: Solange sich niemand beschwert passiert auch nix.

**Comment by cagix** [comment]
> Zwei Versionen haben immer den faden Geschmack von zwei verschiedenen Bugfixes.
> Wir hatten das im alten Dungeon schonmal durchgemacht, war nicht schön.

Genau, u.a. deshalb war ich ursprünglich auch gegen Versionen und/oder Branches.


> Wir erstellen einen Branch wie von dir beschrieben, führen bugfixes aber nur auf Zuruf der Studis aus. Heißt: Solange sich niemand beschwert passiert auch nix.

Das klingt für mich nach einem Plan 👍

**Comment by cagix** [comment]
@fwatermann @AHeinisch Könnt ihr bitte nach dem letzten non-breaking Commit/Merge schauen? Dann würde ich dort den Branch anlegen und an die Studis kommunizieren.

**Comment by AHeinisch** [comment]
Leider habe ich jetzt nicht den Überblick, welche Aufgaben alle existieren. Der PR https://github.com/Programmiermethoden/Dungeon/pull/597 hat, so weit ich das erkennen kann, etwas geändert, wo es um die Ohren fliegen kann. Ich habe jetzt nur 6 Forks angeschaut und dies war meist einer, der noch nicht übernommen wurde. Somit wäre https://github.com/Programmiermethoden/Dungeon/commit/a2e780eb320d28b3f25fc35ae3bf849d00b0791d der letzte non-breaking.

@fwatermann vielleicht kannst du dort einmal kontrollieren, ob dies auch deine Erkenntnis ist.

**Comment by Lena241** [comment]
Ich habe jetzt einen Branch mit dem Namen "pm_s23_master" angelegt  (https://github.com/Programmiermethoden/Dungeon/tree/pm_s23_master) und auf den Stand von https://github.com/Programmiermethoden/Dungeon/commit/a2e780eb320d28b3f25fc35ae3bf849d00b0791d gebracht. Ich habe nicht kontrolliert, ob das jetzt wirklich der letzte non-breaking Stand ist, da habe ich jetzt mal euch beiden vertraut.

**Comment by cagix** [comment]
OK, dann kann das Ticket eigentlich zu, oder?
